src/senaite/core/content/labproduct.py

```diff
@@ -0,0 +1,226 @@
+            default="Total Price"
+        ),
+        readonly=True)
+
+
+@implementer(ILabProduct, ILabProductSchema, IHavePrice, IDeactivable)
+class LabProduct(Item):
+    """LabProduct content
+    """
+    # Catalogs where this type will be catalogued
+    _catalogs = [SETUP_CATALOG]
+
+    security = ClassSecurityInfo()
+
+    @security.protected(permissions.View)
+    def getVolume(self):
+        accessor = self.accessor("labproduct_volume")
+        value = accessor(self) or ""
+        return api.to_utf8(value)
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setVolume(self, value):
+        mutator = self.mutator("labproduct_volume")
+        mutator(self, api.safe_unicode(value))
+
+    # BBB: AT schema field property
+    Volume = property(getVolume, setVolume)
+
+    @security.protected(permissions.View)
+    def getUnit(self):
+        accessor = self.accessor("labproduct_unit")
+        value = accessor(self) or ""
+        return api.to_utf8(value)
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setUnit(self, value):
+        mutator = self.mutator("labproduct_unit")
+        mutator(self, api.safe_unicode(value))
+
+    # BBB: AT schema field property
+    Unit = property(getUnit, setUnit)
+
+    @security.protected(permissions.View)
+    def getPrice(self):
+        accessor = self.accessor("labproduct_price")
+        value = accessor(self) or ""
+        return api.to_utf8(str(value))
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setPrice(self, value):
+        mutator = self.mutator("labproduct_price")
+        mutator(self, api.safe_unicode(value))
+
+    # BBB: AT schema field property
+    Price = property(getPrice, setPrice)
+
+    @security.protected(permissions.View)
+    def getVAT(self):
+        accessor = self.accessor("labproduct_vat")
+        value = accessor(self) or ""
+        return api.to_utf8(str(value))
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setVAT(self, value):
+        mutator = self.mutator("labproduct_vat")
+        mutator(self, api.safe_unicode(value))
+
+    # BBB: AT schema field property
+    VAT = property(getVAT, setVAT)
+
+    def getDefaultVAT(self):
+        """ return default VAT from setup """
+        return api.get_setup().getVAT() or "0.00"
+
+    @security.protected(permissions.View)
+    def getVATAmount(self):
+        """ Compute VATAmount
+        """
+        try:
+            vatamount = self.getTotalPrice() - Decimal(self.getPrice())
+        except Exception:
+            vatamount = Decimal('0.00')
+        return vatamount.quantize(Decimal('0.00'))
+
+    labproduct_vat_amount = ComputedAttribute(getVATAmount, 1)
+
+    # BBB: AT schema (computed) field property
+    VATAmount = property(getVATAmount)
+
+    def getTotalPrice(self):
+        """ compute total price """
+        price = Decimal(self.getPrice() or '0.00')
+        vat = Decimal(self.getVAT())
```

Comment:
Here you do already the conversion to `Decimal`, which is the way how we want to do it.
In the DB we store strings and convert it to a `float` or `Decimal` when we want to do calculations with it.